File: csharp/pascals-triangle/PascalsTriangle.cs
public static class PascalsTriangle
{
    public static IEnumerable<IEnumerable<int>> Calculate(int rows)
    {
        List<List<int>> triangle = new List<List<int>>();
        for (int i = 0; i < rows; i++)
        {
            List<int> numbers = new();
            for (int j = 0; j <= i; j++)
            {
                if (j == 0 || j == i)
                {
                    numbers.Add(1);
                }
                else
                {
                    numbers.Add(triangle[i - 1][j - 1] + triangle[i - 1][j]);
                }
            }
            triangle.Add(numbers);
        }
        return triangle;
    }

}

File: typescript/pascals-triangle/pascals-triangle.ts
export class Triangle {
    private triangle: number[][] = [];
    constructor(count: number) {
        for (let i = 0; i < count; i++) {
            const row: number[] = [1];
            for (let j = 1; j < i; j++) {
                row.push(this.triangle[i - 1][j - 1] + this.triangle[i - 1][j]);
            }
            if (i > 0) {
                row.push(1);
            }
            this.triangle.push(row);
        }
    }
    get rows(): number[][] {
        return this.triangle;
    }
    get lastRow(): number[] {
        return this.triangle[this.triangle.length - 1] || [];
    }
}
